### PR TITLE
Compact compiler buttons

### PR DESCRIPTION
**Pull request by mattgodbolt**
Requesting a PR from Jared as this touches `static` and I don't want to cause any upset :)

### CODE REVIEW

**Comment by jaredwy** [comment]
looks ok to me (although js solutions to css problems make me twitch :D) 
At this stage just anything modifying includes will hurt.

**Comment by mattgodbolt** [comment]
Right! If you know of a way to solve this in pure CSS I'm all ears!

**Comment by jaredwy** [comment]
for css you could find the point at which things break, record the screen width in devtools. Use a media query to change at that width .hideable { display: none }

**Comment by mattgodbolt** [comment]
@jaredwy that works for the screen size being that small. This monitors the size of a particular pane in the view, and if it's too small, hides the long elements. Unless it would spill onto two lines anyway, in which case there's no benefit to making it smaller... that's why the JS is there. TBH if I could just hide if an element was <= some size I'd do that, but as I had to crack out a JS solution anyway I thought I'd try and be "clever".

**Comment by jaredwy** [comment]
Ah, so there isn't a correlation between screen size and pane size?

**Comment by mattgodbolt** [comment]
Only that pane size <= screen size. The panes are the resizable panels (the editor, the compiler) so their size can be changed by resizing the whole browser window, or by dragging the edges of the panes to resize them.

File: static/compiler.js [hunk 1]
```diff
@@ -271,7 +271,19 @@ Compiler.prototype.undefer = function () {
 // Issue manifests if you make a window where one compiler is small enough that the buttons spill onto two lines:
 // reload the page and the bottom-bar is off the bottom until you scroll a tiny bit.
 Compiler.prototype.resize = function () {
-    var topBarHeight = this.topBar.outerHeight(true);
+    // If we save vertical space by hiding stuff that's OK to hide
+    // when thin, then hide that stuff.
+    var hideable = this.domRoot.find('.hideable');
```
[the remark this answers is not in the document]
Comment (mattgodbolt) [reply static/compiler.js]: I considered this, but resize doesn't happen all that often, and the complexity of the caching logic seemed unnecessary...but it's easy to do and so I will :) thanks for the feedback.

[the remark this answers is not in the document]
Comment (jaredwy) [reply static/compiler.js]: Resize doesn't happen to often but it seems to fire a lot when resize is underway so querying the dom many times a second didn't seem great :)

[the remark this answers is not in the document]
Comment (mattgodbolt) [reply static/compiler.js]: I guess so, yes. Though if you're resizing the width is probably changing anyway. It's still worth it though!

[the remark this answers is not in the document]
Comment (mattgodbolt) [reply static/compiler.js]: Added a fix, please take another look!

**Comment by jaredwy** [comment]
👍 lgtm

**Comment by mattgodbolt** [comment]
Thanks :grinning: - always good to have this kind of feedback!

File: static/compiler.js [hunk 2]
```diff
@@ -73,6 +73,7 @@ function Compiler(hub, container, state) {
     this.compilerService = hub.compilerService;
     this.domRoot = container.getElement();
     this.domRoot.html($('#compiler').html());
+    this.hideable = this.domRoot.find('.hideable');
```
Comment (RabsRincon) [inline static/compiler.js]: This might be better off on `Editor.initButtons`

Comment (mattgodbolt) [reply static/compiler.js]: Good spot! Thanks!

**Pull request merged**